python/pyspark/sql/dataframe.py

```diff
@@ -1982,6 +1982,8 @@ def dropDuplicates(self, subset=None):
         """
         if subset is None:
             jdf = self._jdf.dropDuplicates()
+        elif not isinstance(subset, list):
+            raise TypeError("Parameter 'subset' must be a list of columns")
```

Comment:
wouldn't it be better to use something like what we have in dropna ?
i.e.
`
        elif isinstance(subset, str):
            subset = [subset]
        elif not isinstance(subset, (list, tuple)):
            raise TypeError("subset should be a list or tuple of column names")
`

Reply:
Let's also have a test by catching exception class and messages.

Reply:
added the tests and the better error checking